app/models/job_execution.rb

```diff
@@ -95,7 +104,6 @@ def run!(&block)
     @job.update_output!(OutputAggregator.new(@output).to_s)
     @subscribers.each(&:call)
```

Comment:
FYI, we should probably wrap each call in a try/catch block.  I've seen instances where there jobs in the global registry that completed, but were never removed because some code in one of the subscribers raised an exception, and the remaining subscriptions (and any code below this point) was never called.